### ISSUE TITLE
Gates as power-ups and resources

### ISSUE THREAD

**Issue by HuangJunye** [HuangJunye/QPong-Unity#7]
I have an idea to make the gates as resources in the game. For example, X gate is free and unlimited, but H gate needs to be earned. We can start with allowance of 1 H gate. The player earns 1 H gate when the ball is bounce back from paddle (with only X gates) The superposition of paddle can be transferred to the ball. The use of H gate creates multiple ball superpositions which will reduce the chance of catching the balls of the classical paddle. 
Along this line, perhaps we can add CNOT gate as a more powerful power up, which costs 3 H gates. We need to come up with the effects of CNOT gates.

**Comment by boland25** [comment]
hmmm, i like it.  will take some figuring out to do it right though.  can bring this up for discussion in the group meeting later.  see if anyone has any ideas

**Comment by boland25** [comment]
@HuangJunye I spoke with some others about how to make the game a better experience.  Everyone was in agreement to keep it simple to play.  Limit the gates to X & H.  Limit the amount of gates you can add to a line to 3.  so 3 qubits, and 3 columns.  use the rest of the space for some sort of instruction is possible.  This might be a good idea for version 1, and worry about the other gates like CNOT in version 2

**Comment by HuangJunye** [comment]
@boland25 I agree to simply as much as possible. For this game, Y is the same as X, and Z doesn't do anything. CNOT is very similar to H gates too. 3x3 gates is sufficient to make any sorts of circuits that the game needs. I think rotation gates might be cool to add, in addition to X and H gates, but we can add that in version 2.
